google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java

```diff
@@ -905,6 +917,8 @@ TransactionContextImpl newTransaction() {
     }
     
     <T extends SessionTransaction> T setActive(@Nullable T ctx) {
+      throwIfTransactionsPending();
```

Comment:
We need a way to disable throwing an exception here in case the user happens to have a legitimate case for running a transaction within another transaction. This could be an option passed to SpannerOptions or settable in the TransactionContext.

Reply:
Thanks. Added a flag to control the behaviour.